src/jmclient/wallet.py

```diff
@@ -845,10 +868,19 @@ def get_balance_by_mixdepth(self, verbose=True,
         confirmations, set maxheight to max acceptable blockheight.
         returns: {mixdepth: value}
         """
+        balances = collections.defaultdict(int)
+        for md in range(self.mixdepth + 1):
+            balances[md] = self.get_balance_at_mixdepth(md, verbose=verbose,
+                include_disabled=include_disabled, maxheight=maxheight)
+        return balances
```

Comment:
I believe this exists to handle mixdepth changes/increases of the wallet. But it's been a long time since I wrote the code.

Reply:
But since we are assigning to it, not accessing the value, there should be no difference? Maybe there are some cases I'm missing.

Reply:
The defaultdict is returned by the function. In order to reason about it you'd have to check the calling code.

Reply:
Okay. If that's the case that callers need this it might be good to spell this out in the docstring that this intentionally return a `defaultdict` to save the caller a check when searching for missing mixdepths.